Rover/Log.cpp

```diff
@@ -38,46 +38,53 @@ void Rover::Log_Write_Attitude()
 // Write a range finder depth message
 void Rover::Log_Write_Depth()
 {
-    // only log depth on boats with working downward facing range finders
-    if (!rover.is_boat() || !rangefinder.has_data_orient(ROTATION_PITCH_270)) {
+    // only log depth on boats
+    if (!rover.is_boat()) {
         return;
     }
 
     // get position
     Location loc;
     IGNORE_RETURN(ahrs.get_position(loc));
 
-    // check if new sensor reading has arrived
-    uint32_t reading_ms = rangefinder.last_reading_ms(ROTATION_PITCH_270);
-    if (reading_ms == rangefinder_last_reading_ms) {
-        return;
-    }
-    rangefinder_last_reading_ms = reading_ms;
-
-    // get temperature
-    float temp_C;
-    if (!rangefinder.get_temp(ROTATION_PITCH_270, temp_C)) {
-        temp_C = 0.0f;
+    for (uint8_t i=0; i<RANGEFINDER_MAX_INSTANCES; i++) {
+        AP_RangeFinder_Backend *s = rangefinder.get_backend(i);
+        
+        if (s == nullptr || s->orientation() != ROTATION_PITCH_270 || !s->has_data()) {
+            continue;
+        }
+
+        // check if new sensor reading has arrived
+        const uint32_t reading_ms = s->last_reading_ms();
+        if (reading_ms == rangefinder_last_reading_ms[i]) {
+            continue;
+        }
+        rangefinder_last_reading_ms[i] = reading_ms;
+
+        float temp_C;
+        temp_C = s->get_temp(temp_C) ? temp_C : 0.0f; 
```

Comment:
total nitpick. This has an extra unnecessary assignment right? Compared to just the if()
```suggestion
        temp_C = s->get_temp(temp_C) ? temp_C : 0.0f; 
```

Reply:
Oh. Got it.